src/app/common/components/presentation/text-box/text-box.component.html

```diff
@@ -1 +1,7 @@
-<input type="text" [id]="id" [placeholder]="placeholder" [value]="value" (click)="click($event)" />
+<div class="form-template">
+  <mat-form-field [color]="warn">
+    <input matInput placeholder={{placeholder}} [type]="hide ? 'password' : 'text'">
```

Comment:
mutliple event emitters need to be exposed here, onkeyup and etc... Also, there needs to be a way to describe a model that can be passed from a parent.